yarn-project/cli/aztec-cli

```diff
@@ -115,39 +112,19 @@ if [[ "$AZTEC_CLI_COMMAND" == "unbox" ]]; then
   add_mount "$DIR"
 fi
 
-# process flags
-if [[ "$AZTEC_CLI_COMMAND" == "compile" || "$AZTEC_CLI_COMMAND" == "call" || "$AZTEC_CLI_COMMAND" == "send" ]]; then
-
-  # bash's builtin getops only works with single characeter flags
-  # GNU getopt doesn't exist on macOS
-  # process the flags manually
-  # NOTE: this won't work with assignement-style flags, e.g. --outdir=/foo
-  for (( i=2; i <= "$#"; i++ )); do
-      arg_value=${!i}
-      next_index=$((i+1))
-      # edge case: odd number of flags
-      if [[ "$next_index" -gt "$#" ]]; then
-          continue
-      fi
-      next_arg=${!next_index}
-      case $arg_value in
-          -o | --outdir)
-              add_mount "$next_arg"
-              ;;
-          -ts | --typescript)
-              add_mount "$next_arg"
-              ;;
-          -i | --interface)
-              add_mount "$next_arg"
-              ;;
-          -c | --contract-artifact)
-              add_mount "$next_arg"
-              ;;
-          *)
-              ;;
-      esac
-  done
-fi
+# bash's builtin getops only works with single characeter flags
+# GNU getopt doesn't exist on macOS
+# process the flags manually
+#
+# go through each parameter (execpt the first one, which is the command)
+# and check if it's either a filename or a directory. If it is then mount inside the container
+# NOTE: this won't work with assignement-style flags, e.g. --outdir=/foo
+for (( i=2; i <= "$#"; i++ )); do
+    arg_value=${!i}
+    if [[ -f "$arg_value" || -d "$arg_value" ]]; then
+      add_mount "$arg_value"
+    fi
```

Comment:
Technically `add_mount` already checks that its argument is an existing directory so this `if` statement could be skipped.